app/client/cli/utils.go

```diff
@@ -0,0 +1,146 @@
+package cli
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/pokt-network/pocket/app/pocket/rpc"
+	"github.com/pokt-network/pocket/shared/crypto"
+	sharedTypes "github.com/pokt-network/pocket/shared/types"
+	utilityTypes "github.com/pokt-network/pocket/utility/types"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// readEd25519PrivateKeyFromFile returns an Ed25519PrivateKey from a file where the file simply encodes it in a string (for now)
+// TODO(pocket/issues/150): this is a temporary hack since we don't have yet a keybase, the next step would be to read from an "ArmoredJson" like in V0
+func readEd25519PrivateKeyFromFile(pkPath string) (pk crypto.Ed25519PrivateKey, err error) {
+	pkFile, err := os.Open(pkPath)
+	if err != nil {
+		return
+	}
+	defer pkFile.Close()
+	pk, err = parseEd25519PrivateKeyFromReader(pkFile)
+	return
+}
+
+func parseEd25519PrivateKeyFromReader(reader io.Reader) (pk crypto.Ed25519PrivateKey, err error) {
+	if reader == nil {
+		return nil, fmt.Errorf("cannot read from reader %v", reader)
+	}
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(reader)
+
+	priv := &crypto.Ed25519PrivateKey{}
+	err = priv.UnmarshalJSON(buf.Bytes())
+	if err != nil {
+		return
+	}
+	pk = priv.Bytes()
+	return
+}
+
+// Credentials reads a password from the prompt and returns the trimmed version
+//
+// If pwd is provided (via flag to the command), it uses that one instead of asking via prompt
+func Credentials(pwd string) string {
+	if pwd != "" && strings.TrimSpace(pwd) != "" {
+		return strings.TrimSpace(pwd)
+	} else {
+		bytePassword, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		if err != nil {
+			fmt.Println(err)
```

Comment:
This is the perfect example to explain what I mean. I know it's a minor thing and we shouldn't spend too much time on this anyway, I just want to get my thought across.

I get this from the `log` package: 
![image](https://user-images.githubusercontent.com/29378614/189062147-8e460829-b7cf-4b07-9935-2999386dcf81.png)

But the devil is in the detail 😈 and what it means that it's a call to `log.Printf()` (emphasis on `log`) followed by `os.Exit()`

`log.Printf()` and `fmt.Printf()`, despite having the same signature, behave slightly differently (hence the separate package).

### log.Printf()
![image](https://user-images.githubusercontent.com/29378614/189063474-ab5a04c8-e0d9-45c6-8e72-1f41f867c9cb.png)
![image](https://user-images.githubusercontent.com/29378614/189063508-01b0c6e6-a03a-4980-9e28-22f9df27b047.png)

### fmt.Printf()
![image](https://user-images.githubusercontent.com/29378614/189063742-a9ca9070-26f8-4870-b243-73c0ac4036d6.png)
![image](https://user-images.githubusercontent.com/29378614/189063874-bbece7e9-a263-46a5-a663-97b0ade46f68.png)

If a long-running process, like the node for example, fails. **YES**, I am expecting a timestamp in the output and to log since these logs could also be digested by some aggregator down the line.

If instead, I am running a command that is short-lived and gives me immediate feedback like in the CLI, I know what time it is and I am not logging anything really.

The same applies to other instances of `PrintLn` or `Printf` (there are just a couple of them in total in the CLI as we speak but obviously there will be more in the future).

I am not marrying `fmt` or hating `log`, I am just trying to learn what's the best approach and sorry if you have to pay for my tuition with your time 😝

Let's also be pragmatic, happy to take this offline and just go with the suggestion to push this. We can always change later if needed, just flagging.

Addressed in https://github.com/pokt-network/pocket/pull/177/commits/d258fd576ae359c994f9cac8df5a56408d9b271b

Reply:
**TIL!!!**

Again, thank you for the explanation and for helping me learn. Given what you said, I think we should 💍 `fmt.Printf` in the context of the CLI :D

Reply:
This is now tracked in #200